packages/transaction-metrics-exporter/src/block-processor.ts

```diff
@@ -0,0 +1,266 @@
+      await Promise.all([this.fetchBlockState(blockNumber), this.processBlockHeader(blockNumber)])
+    }
+  }
+
+  async fetchBlockState(blockNumber: number) {
+    const promises = stateGetters.map(
+      async ({ contract, method, args, transformValues, maxBucketSize }) => {
+        if (!(await this.contracts[contract].exists(blockNumber))) {
+          return
+        }
+        args = typeof args === 'function' ? args(blockNumber) : args
+        this.contracts[contract].setDefaultBlock(blockNumber)
+        return (this.contracts as any)[contract][method](...args)
+          .then((returnData: any) => {
+            this.contracts[contract].setDefaultBlock('latest')
+            const data: DataResult = {
+              contract,
+              function: method,
+              args: JSON.stringify(args),
+              blockNumber,
+              values: transformValues(returnData),
+            }
+            this.logEvent(LoggingCategory.State, data)
+            this.logHistogram({ ...data, args }, maxBucketSize)
+          })
+          .catch(() => '') as Promise<any>
+      }
+    )
+    await Promise.all(promises)
+  }
+
+  async processBlockHeader(blockNumber: number) {
+    const NOT_WHITELISTED_ADDRESS = 'not_whitelisted_address'
+
+    const blockExplorer = await newBlockExplorer(this.kit)
+    const logExplorer = await newLogExplorer(this.kit)
+
+    const block = await blockExplorer.fetchBlock(blockNumber)
+    const previousBlock: Block = await blockExplorer.fetchBlock(blockNumber - 1)
+
+    Counters.blockheader.inc({ miner: block.miner })
+
+    const blockTime = Number(block.timestamp) - Number(previousBlock.timestamp)
+    this.logEvent(LoggingCategory.Block, { ...block, blockTime, blockNumber })
+
+    const parsedBlock = blockExplorer.parseBlock(block)
+    const parsedTxMap = toTxMap(parsedBlock)
+
+    for (const tx of parsedBlock.block.transactions as Transaction[]) {
+      const parsedTx: ParsedTx | undefined = parsedTxMap.get(tx.hash)
+
+      this.logEvent(LoggingCategory.Transaction, tx)
+      const receipt = await this.kit.web3.eth.getTransactionReceipt(tx.hash)
+      this.logEvent(LoggingCategory.TransactionReceipt, receipt)
+      ;(this.kit.web3.currentProvider as any).existingProvider.send(
+        {
+          method: 'debug_traceTransaction',
+          params: [
+            tx.hash,
+            {
+              tracer: tracerAsText,
+              disableStack: true,
+              disableMemory: true,
+              disableStorage: true,
+            },
+          ],
+          jsonrpc: '2.0',
+          id: '2',
+        },
+        (err: any, result: any) => {
+          if (!err) {
+            result?.result
+              ?.filter?.(
+                ({ type, callType }: any) => type === 'create' || callType === 'delegatecall'
+              )
+              ?.forEach?.((data: any) =>
+                this.logEvent(LoggingCategory.InternalTransaction, {
```

Comment:
What exactly does this log? I thought we would try to parse the internal transaction as well?

Reply:
It is exactly what we do. This line is logging each of the internal transactions that have been parsed previously.